articles/quickstart/webapp/any/01-login.md

```diff
@@ -0,0 +1,178 @@
+
+In any case, Auth0 provides back to your application the authenticated identity of the user --- a unique third-party profile or email address --- and it is up to your application to respond to that identity apporopriately. 
+
+## Register Your App as a Client
+
+Go to the [Clients](${manage_url}/#/clients) page from the [Auth0 Dashboard](${manage_url}). Edit the default client or create a new client by clicking **Create Client**.
+
+Name your new application, and set the Client Type to **Regular Web Applications** as the **Client Type**.
+
+### Set Allowed Callback URI
+
+When your app redirects a user to Auth0, the querystring will include a callback URI. After authentication completes, the user will be redirected to this URI.
+
+${include('../_callbackRegularWebApp')}
+
+While you are editing your Client settings, update your Allowed Callbacks list to include the redirect URI you plan to use.
+
+### Enable OIDC Conformance
+
+Before you leave Client Setting, click on **Show Advanced Settings**. Go to the **OAuth** tab and make sure that the **OIDC Conformant** switch is enabled.
+
+Save your settings.
+
+## Create a Log In Link
+
+Your users will log in to your site using the Auth0 authentication portal at the URI `https://${account.namespace}/authorize`.
+
+To facilitate this, you will provide a link from your site with the following query parameters:
+
+| Parameter | Value |
+|:----------|:---------|
+| response_type | `code` |
+| client_id | ${account.clientId} |
+| scope | `openid` |
+| redirect_uri | The URI that the user will be redirected to after authentication (`${account.callback}`)|
+| state | YOUR_OPAQUE_VALUE
+
+```html
+<a href="https://${account.namespace}/authorize?response_type=code&client_id=${account.clientId}&scope=openid&redirect_uri=${account.callback}&state=YOUR_OPAQUE_VALUE">Log In</a>
+```
+
+### State Value
+
+The `state` is a nonce, or arbitrary value, which the client provides and which the Auth0 service includes in its redirect.
+This helps guard against cross-site request forgery. 
+
+Your application should create this value, store in the browser, and compare it to the state value received in the callback querystring. If they don't match, you should deny access to the user.
+
+[Learn more here.](/protocols/oauth2/oauth-state)
+
+
+## Handle the Callback
+
+After the user has authenticated, Auth0 will redirect the user to the URI specified in the `redirect_uri` parameter of the query string. A query string will be added to the redirect URI, providing a code. 
+
+```text
+${account.callback}?code=tQPUv...
+```
+
+Your application must then handle the request appropriately.
+
+Typically, this means:
+
+ - catch the request as it comes in
+ - extract the value of the `code` parameter in the quesrystring
+ - call `/oauth/token` endpoint to exchange the code for an id_token
+ - decode the id_token, identifying the user
+ - responding the request appropriately, based on the verified identity of the user.
+ 
```

Comment:
Should the callback do something regarding the 'state' parameter?

Reply:
Yes. I mentioned it above, but it should be repeated here since this is when it happens.